plugin/adsk/plugin/adskMaterialCommands.cpp

```diff
@@ -0,0 +1,180 @@
+        = std::find(sourceTypes.cbegin(), sourceTypes.cend(), TfToken("arnold"))
+        != sourceTypes.cend();
+
+    if (hasArnoldMaterials) {
+        // TODO: Replace hard-coded materials with dynamically generated list.
+        const MString label = "AI Standard Surface";
+        const MString identifier = "arnold:standard_surface";
+        appendToResult("Arnold/" + label + "|" + identifier);
+    }
+}
+
+void ADSKMayaUSDGetMaterialsForRenderersCommand::appendUsdMaterials() const
+{
+    // TODO: Replace hard-coded materials with dynamically generated list.
+    const MString label = "USD Preview Surface";
+    const MString identifier = "UsdPreviewSurface";
+    appendToResult("USD/" + label + "|" + identifier);
+}
+
+// main MPxCommand execution point
+MStatus ADSKMayaUSDGetMaterialsForRenderersCommand::doIt(const MArgList& argList)
+{
+    clearResult();
+
+    MStatus status;
+    if (!status)
+        return status;
+
+    // TODO: The list of returned materials is currently hard-coded and only for select,
+    // known renderers. We should populate the material lists dynamically based on what the
+    // installed renderers report as supported materials.
+    appendUsdMaterials();
+    appendArnoldMaterials();
+    appendMaterialXMaterials();
+
+    return MS::kSuccess;
+}
+
+MSyntax ADSKMayaUSDGetMaterialsForRenderersCommand::createSyntax()
+{
+    MSyntax syntax;
+    return syntax;
+}
+
+/*
+// ADSKMayaUSDGetMaterialsInStageCommand
+*/
+
+const char ADSKMayaUSDGetMaterialsInStageCommand::commandName[] = "mayaUsdGetMaterialsInScene";
+
+static const TfToken materialType("Material");
+
+// plug-in callback to create the command object
+void* ADSKMayaUSDGetMaterialsInStageCommand::creator()
+{
+    return static_cast<MPxCommand*>(new ADSKMayaUSDGetMaterialsInStageCommand());
+}
+
+// private argument parsing helper
+MStatus ADSKMayaUSDGetMaterialsInStageCommand::parseArgs(const MArgList& argList)
+{
+    return MS::kSuccess;
+}
+
+// main MPxCommand execution point
+MStatus ADSKMayaUSDGetMaterialsInStageCommand::doIt(const MArgList& argList)
+{
+    clearResult();
+
+    MStatus      status;
+    MArgDatabase args(syntax(), argList, &status);
+    if (!status)
+        return status;
+
+    MString ufePathString;
+    args.getFlagArgument(kSceneItem, 0, ufePathString);
+    if (ufePathString.isEmpty()) {
+        MGlobal::displayError("Missing argument 'sceneItem'.");
+        return MS::kFailure;
+    }
+
+    const auto  ufePath = Ufe::PathString::path(ufePathString.asChar());
+    UsdStagePtr stage = ufe::getStage(ufePath);
+    if (stage) {
+        for (auto prim : stage->Traverse()) {
+            if (prim.GetTypeName() == materialType) {
+                appendToResult(MString(prim.GetPath().GetString().c_str()));
+            }
+        }
+    }
+
+    return MS::kSuccess;
+}
+
+MSyntax ADSKMayaUSDGetMaterialsInStageCommand::createSyntax()
+{
+    MSyntax syntax;
+    syntax.addFlag("-si", "-sceneItem", MSyntax::kString);
```

Comment:
I don't think you need or should use a flag like this. Its not really a "sceneitem". It is actually a string which is interpreted as a Ufe PathString. 

Instead look at calling ```addArg(MSyntax::kString)``` then simply get that string arg. There are examples of this in our code already. Plus you should probably set the ```syntax.enableQuery/enableEdit``` as well.